### PR TITLE
Fix TS minimap terrain colours

### PR DESCRIPTION
**Pull request by pchote**
My original interpretation was quite wrong... I don't know what I was thinking 😅 
From the commit message:
```
* Rename LeftColor and RightColor to MinColor and MaxColor
  These are mapped from LowRadarColor and HighRadarColor in
  the original inis, and appear to be used to set the bounding
  values for selecting a random colour, NOT for left/right
  pixels (which caused noticeably wrong banding).

* Adjust brightness based on terrain height.
  MinHeightColorBrightness and MaxHeightColorBrightness
  were chosen by trial/error to match the original
  map preview rendering.
```
Thanks to tomsons26 for explaining that the original games hardcode tileset-specific brightness bodges in the exe. I don't know what those bodges actually *are*, so I picked modifiers that matched as closely as I could tell by eye.

An obvious scope creep will be to replace `TerrainTypeInfo.Color` with `MinColor`/`MaxColor` and define appropriate ranges for RA/TD/D2k's terrain types to add visual noise and break up the large constant color blocks there too. That will mean regenerating all of the upstream oramaps, so I would rather not do that here where the focus is on TS.

Some before/original TS/after comparisons:

arivruns-old
arivruns-minimap
arivruns-new

float-old
float-minimap
float-new

tread_l-old
tread_l-minimap
tread_l-new

### CODE REVIEW

File: OpenRA.Mods.Common/UpdateRules/Rules/20200503/UpdateTilesetColors.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	class UpdateTilesetColors : UpdateRule
+	{
+		public override string Name { get { return "Rename Tileset LeftColor and RightColor"; } }
+		public override string Description
+		{
+			get
+			{
+				return "The LeftColor and RightColor keys in isometric tilesets have been renamed to MinColor and MaxColor to reflect their proper usage.";
```
Comment (Mailaender) [inline OpenRA.Mods.Common/UpdateRules/Rules/20200503/UpdateTilesetColors.cs]: Although maybe not intended you can also use them in rectangular maps. https://github.com/OpenHV/OpenHV/commit/da97b59f4eeaf4701cfe8fc492a7e14edd46075a

image

where they give an illusion of depth without me having to define yet another terrain type.

File: OpenRA.Game/Map/Map.cs [hunk 2]
```diff
@@ -716,8 +722,21 @@ public byte[] SavePreview()
...
+									var right = Exts.ColorLerp(Game.CosmeticRandom.NextFloat(), type.MinColor, type.MaxColor);
+									var scale = float2.Lerp(minHeightBrightness, maxHeightBrightness, heightModifier * Height[uv]);
+									leftColor = Color.FromArgb((int)(scale * left.R).Clamp(0, 255), (int)(scale * left.G).Clamp(0, 255), (int)(scale * left.B).Clamp(0, 255));
+									rightColor = Color.FromArgb((int)(scale * right.R).Clamp(0, 255), (int)(scale * right.G).Clamp(0, 255), (int)(scale * right.B).Clamp(0, 255));
+								}
+								else
+									leftColor = rightColor = type.MinColor;
+							}
+							else
+								leftColor = rightColor = Color.Black;
```
Comment (Mailaender) [inline OpenRA.Game/Map/Map.cs]: Shouldn't this be

```suggestion
								else
								{
									leftColor = type.MinColor;
									rightColor = type.MaxColor;
								}
							}
							else
								leftColor = rightColor = Color.Black;
```

?

Comment (pchote) [reply OpenRA.Game/Map/Map.cs]: No. The two color values do not correspond to the left and right pixels in the isometric minimaps - this was bogus in the previous implementation.

**Comment by Mailaender** [comment]
You forgot about https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Cnc/UtilityCommands/LegacyTilesetImporter.cs but otherwise the result speaks for itself. It looks way better.

**Comment by pchote** [comment]
Fixed.

File: OpenRA.Game/Map/Map.cs [hunk 3]
```diff
@@ -17,6 +17,7 @@
 using System.Text;
 using OpenRA.FileFormats;
 using OpenRA.FileSystem;
+using OpenRA.Graphics;
```
Comment (teinarss) [inline OpenRA.Game/Map/Map.cs]: not needed.

File: OpenRA.Game/Map/Map.cs [hunk 4]
```diff
@@ -716,8 +722,21 @@ public byte[] SavePreview()
...
+								if (useHeightModifiers)
+								{
+									var left = Exts.ColorLerp(Game.CosmeticRandom.NextFloat(), type.MinColor, type.MaxColor);
+									var right = Exts.ColorLerp(Game.CosmeticRandom.NextFloat(), type.MinColor, type.MaxColor);
+									var scale = float2.Lerp(minHeightBrightness, maxHeightBrightness, heightModifier * Height[uv]);
+									leftColor = Color.FromArgb((int)(scale * left.R).Clamp(0, 255), (int)(scale * left.G).Clamp(0, 255), (int)(scale * left.B).Clamp(0, 255));
+									rightColor = Color.FromArgb((int)(scale * right.R).Clamp(0, 255), (int)(scale * right.G).Clamp(0, 255), (int)(scale * right.B).Clamp(0, 255));
+								}
+								else
+									leftColor = rightColor = type.MinColor;
```
Comment (teinarss) [inline OpenRA.Game/Map/Map.cs]: Extract this code and reuse it in `PlayerRadarTerrain.cs ` to remove duplication?

**Comment by pchote** [comment]
Updated.

**Review by obrakmann — APPROVED** [review]
:+1:

**Pull request merged**